react/ContactsListModal/index.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from 'react'
+import { Query, fetchPolicies } from 'cozy-client'
+import ContactsList from '../ContactsList'
+import Modal, { ModalHeader, ModalDescription, ModalBackButton } from '../Modal'
+import Spinner from '../Spinner'
+import styles from './styles.styl'
+import Input from '../Input'
+import PropTypes from 'prop-types'
+import withBreakpoints from '../helpers/withBreakpoints'
+
+const thirtySeconds = 30000
+const olderThan30s = fetchPolicies.olderThan(thirtySeconds)
+
+const mkFilter = filter => contacts => {
+  if (!filter) {
+    return contacts
+  }
+
+  const f = filter.toLowerCase()
+
+  return contacts.filter(contact => {
+    return (
+      (contact.fullname && contact.fullname.toLowerCase().includes(f)) ||
+      (contact.name &&
+        contact.name.familyName &&
+        contact.name.familyName.toLowerCase().includes(f)) ||
+      (contact.name &&
+        contact.name.givenName &&
+        contact.name.givenName.toLowerCase().includes(f))
+    )
+  })
```

Comment:
We can have contact having only email's address or cozy-url (for instance if you have shared something with it) 

In drive we use a contact filter for the sharing process. Few methods are here: https://github.com/cozy/cozy-drive/blob/master/src/sharing/suggestionMatchers.js 

Since we're in the process of exporting `sharings` to cozy-libs, maybe we should also export search / filter to cozy-doctypes at least. 

Also, as @y-lohse will say : `I prefer using lodash.get instead of having to do several checks`

Reply:
I'm filtering on `Contact.getDisplayName` now, which handles contacts with only email address. I added a TODO comment to extract matchers functions from drive later.